.github/workflows/build_test.yml

```diff
@@ -0,0 +1,62 @@
+name: Build and test
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+    paths-ignore:
+      - '**.rst'
+      - 'docs/**'
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - '**.rst'
+      - 'docs/**'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+  
+jobs:
+  build_test:
+    strategy:
+      matrix:
+        os: ["ubuntu-20.04", "macos-latest"]
+        python: ["3.8", "3.9", "3.10", "3.11"]
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python ${{ matrix.python }}
```

Comment:
Can we install Ruff and test the linter as well in a different workflow?

Reply:
I can split the workflow into different pieces, sure.